File: 001-020/018.rb
nums = []
# 問題を読み込む
File.open('./001-020/018.data.txt') do |f|
  f.each do |line|
    nums << line.split(' ').map(&:to_i)
  end
end
# 下から順に値の大きい方だけ足していく
(nums.length - 2).downto(0) do |row|
  nums[row].length.times do |col|
    nums[row][col] += [nums[row + 1][col], nums[row + 1][col + 1]].max
  end
end

puts nums[0][0]

File: 001-020/018.js
//ファイルからデータを読み込む
let fs = require('fs');
let input_raw = fs.readFileSync('./001-020/018.data.txt', 'utf-8');
// 改行コードのCRLF or CR or LFで分割する
let input = input_raw.split(/\r\n|\r|\n/);
let nums = input.map(x => x.split(' ').map(y => parseInt(y, 10)));

for (let row = nums.length - 2; row >= 0; row--) {
  // for..in でループを書くと変数にNumber型でなく、String型が代入されることがあるので注意
  for (let col = 0; col < nums[row].length; col++) {
    nums[row][col] += Math.max(nums[row + 1][col], nums[row + 1][col + 1]);
  }
}

console.log(nums[0][0]);
